graph/pull.go

```diff
@@ -352,9 +352,9 @@ func (s *TagStore) pullRepository(r *registry.Session, out io.Writer, repoInfo *
 		}
 	}
 
-	requestedTag := repoInfo.CanonicalName
+	requestedTag := repoInfo.LocalName
 	if len(askedTag) > 0 {
-		requestedTag = utils.ImageReference(repoInfo.CanonicalName, askedTag)
```

Comment:
It seemed wrong to me to use the Canonical name when trying to represent the "asked" or "requested" tag. I figured the user's representation is what is most appropriate. Before this change the canonical name was always set to local name so it was not noticeable. Displaying the canonical name instead of the local name still may feel weird, as that will be something very visible to users and possibly confusing. This is one of the reasons I wanted to submit this PR separately, so we can discuss it on its own without it getting mingled into some of the other changes and refactors. WDYT?